frame/cere-staking/src/lib.rs

```diff
@@ -0,0 +1,599 @@
+	/// rounds.
+	#[codec(compact)]
+	pub active: Balance,
+	/// Any balance that is becoming free, which may eventually be transferred out of the stash
+	/// (assuming it doesn't get slashed first). It is assumed that this will be treated as a first
+	/// in, first out queue where the new (higher value) eras get pushed on the back.
+	pub unlocking: BoundedVec<UnlockChunk<Balance>, MaxUnlockingChunks>,
+}
+
+impl<AccountId, Balance: HasCompact + Copy + Saturating + AtLeast32BitUnsigned + Zero>
+	StakingLedger<AccountId, Balance>
+{
+	/// Initializes the default object using the given stash.
+	pub fn default_from(stash: AccountId) -> Self {
+		Self { stash, total: Zero::zero(), active: Zero::zero(), unlocking: Default::default() }
+	}
+
+	/// Remove entries from `unlocking` that are sufficiently old and reduce the
+	/// total by the sum of their balances.
+	fn consolidate_unlocked(self, current_era: EraIndex) -> Self {
+		let mut total = self.total;
+		let unlocking: BoundedVec<_, _> = self
+			.unlocking
+			.into_iter()
+			.filter(|chunk| {
+				if chunk.era > current_era {
+					true
+				} else {
+					total = total.saturating_sub(chunk.value);
+					false
+				}
+			})
+			.collect::<Vec<_>>()
+			.try_into()
+			.expect(
+				"filtering items from a bounded vec always leaves length less than bounds. qed",
+			);
+
+		Self { stash: self.stash, total, active: self.active, unlocking }
+	}
+}
+
+/// Preference of what happens regarding to participating in storage network.
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo, Default)]
+pub struct StoragePrefs {
+	foo: bool, // ToDo
+}
+
+/// Preference of what happens regarding to participating in CDN.
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo, Default)]
+pub struct EdgePrefs {
+	foo: bool, // ToDo
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::{
+		pallet_prelude::*, sp_runtime::traits::StaticLookup, traits::LockableCurrency,
+		Blake2_128Concat,
+	};
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Currency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>;
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		/// Number of eras that staked funds must remain bonded for.
+		#[pallet::constant]
+		type BondingDuration: Get<EraIndex>;
+	}
+
+	/// Map from all locked "stash" accounts to the controller account.
+	#[pallet::storage]
+	#[pallet::getter(fn bonded)]
+	pub type Bonded<T: Config> = StorageMap<_, Twox64Concat, T::AccountId, T::AccountId>;
+
+	/// The minimum active bond to become and maintain the role of a storage network participant.
+	#[pallet::storage]
+	pub type MinStorageBond<T: Config> = StorageValue<_, BalanceOf<T>, ValueQuery>;
```

Comment:
Should it be readable?

Reply:
Sorry, it was my mistake during the review call. Storage items have default querying implementation and getters only affect how they are accessible from within the rust code. A getter is not required here.

https://docs.substrate.io/build/runtime-storage#getter-methods